src/ophys_etl/modules/segmentation/modules/schemas.py

```diff
@@ -465,6 +475,29 @@ class RoiMergerSchema(argschema.ArgSchema):
         description=("fraction of timesteps to keep when doing time "
                      "correlations"))
 
+    @post_load
+    def check_for_rois(self, data, **kwargs):
+        """rois can be passed in as a path to json in 'roi_input'
+        or can be read from within the in-progress 'qc_output'
+        """
+        if data["roi_input"] is None:
```

Comment:
Can we add a check somewhere that raises an error if `data['roi_input']` is not `None` and `qc_output['detect/rois']` exists? I am very worried about users specifying an `roi_input` that is inconsistent with `qc_output`. I think we should do our best to mandate only one source of input ROIs.

Reply:
Given that the merger's plotting block will use the qc ROIs as `original_roi_list` if qc ROIs exist, I think we need to do this.

https://github.com/AllenInstitute/ophys_etl_pipelines/pull/316/files#diff-34c97f6ec7a67ea92ac457f98a76931fe50e7f9cd767e3ab327607f0b71fdb6eR118

Reply:
resolved with major rework. no longer supporting multiple ROI data sources.